### ISSUE TITLE
StartUpDG t8code todo list

### ISSUE THREAD

**Issue by jlchan** [jlchan/StartUpDG.jl#38]
@Davknapp @jmark from our Trixi discussion

1. Generalize connectivity in `MeshData` (remove `mapP`, `mapB` as fields, replace with a general "node connectivity" field to enable `t8code`)
2. Add ability to create connectivities between different `MeshData` objects.

If there is anything else you can think of, please feel free to add those items.

**Comment by jmark** [comment]
@jlchan Can you imaging a `MeshData` constructor which accepts a list of element interfaces (with orientation ;) and per element a list of Jacobians evaluated on quadrature points?

**Comment by jlchan** [comment]
Certainly. I don't need the orientation for my own constructor but I can certainly store it. 

Are the physical points interpolation points or quadrature points? I would prefer interpolation points if possible (easier to enforce a watertight mesh).

**Comment by jmark** [comment]
Sure, interpolation points will work.

Yeah, I think we have to pass the orientation to the user to take care of it so that the interpolated face values from each side correctly match according to the mesh.

**Comment by jlchan** [comment]
Sounds good. FYI, we account for orientation when computing the `mapP` array - any changes in orientation are accounted for in the ordering of the node maps, and these node maps are just computed based on node positions (if two nodes basically have the same coordinates, we map one to the other).
